### PR TITLE
Support Style Dictionary v4 and add preprocessors config

### PR DESCRIPTION
**Pull request by mrajaeim**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,14 +50,14 @@
     "prettier": "2.8.8",
     "rimraf": "3.0.2",
     "semantic-release": "21.1.2",
-    "style-dictionary": "3.9.2",
+    "style-dictionary": "^4.1.0",
```
Comment (nado1001) [inline package.json:53]: It would be great if you could get it to v4.1.1

**Review by nado1001 — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I'm sorry, but is it possible to have the code in the `README.md` also rewritten as follows?

``````js
import StyleDictionary from 'style-dictionary'
import { makeSdTailwindConfig } from 'sd-tailwindcss-transformer'

const StyleDictionaryTailwind = new StyleDictionary(
  makeSdTailwindConfig({ type: 'all' })
)
await StyleDictionaryTailwind.buildAllPlatforms()
```

**Comment by mrajaeim** [comment]
Hi @nado1001,

Thanks for reminding me to update the README. I’ve updated it and also added an integration test to ensure the package works with style-dictionary.

**Review by nado1001 — APPROVED** [review]
Thanks for writing the test as well!

**Review by nado1001 — CHANGES_REQUESTED** [review]
Sorry, there was an error in `src/tests/integration.test.ts`. Could you please fix it?

**Comment by mrajaeim** [comment]
Hi @nado1001 ,

I'm not sure why we're getting a type error in the test for the following line:
`import StyleDictionary from 'style-dictionary'`
This error occurs when running `yarn typecheck` in CI, but the tests pass successfully when running `yarn test`.
It might be a misconfiguration in the TypeScript settings.
Thanks for your review!

**Comment by nado1001** [comment]
Hi, @mrajaeim 
Thanks for the confirmation!
You are right this seems to be caused by the TypeScript configuration.
As it is difficult to solve the problem immediately at the moment, I am sorry, but could you please delete one of the tests you added?
We will add the test you wrote later after we resolve this issue.

**Comment by mrajaeim** [comment]
Hi @nado1001,
I’ve deleted the integration test.

**Review by nado1001 — APPROVED** [review]
@mrajaeim 
Thank you very much!
I'll release it right after the merge.

**Pull request merged**